src/git.md

```diff
@@ -259,8 +258,7 @@ commit. Both the upside and downside of this is that it simplifies the history.
 On the one hand, you lose track of the steps in which changes were made, but
 the history becomes easier to work with.
 
-You also may want to squash just the last few commits together, possibly
-because they only represent "fixups" and not real changes. For example,
+You also may want to squash just the last few commits together. For example,
```

Comment:
Ditto, I think this is good advice that's non-normative. The confusing part of https://github.com/rust-lang/rustc-dev-guide/issues/900 is that it says you *should* in both cases.

Reply:
True, I see. Both changes are in separated commit, do you prefer drop it and force or revert it?

This type of questions are the one that maybe will be wondering someone new, that maybe are not the points that more value give compared with the changes (in other cases, this is only removing a phrase).

Ans also one of the questions of #900:

> Is it OK if the commit strategy depends on the assigned reviewer?

Maybe, apart from the confusing _should_ the review process could need to separate a bit the preferences opinionable.

That only will end with or do with the reviewer preferences learning each reviewer or debates about what each one thing is more beautiful. Instead of arguments about what could produce errors, maintenability, etc.

And maiby frustration if both have strong preferences between what the history must be, ideal commits perfect for one preference or near to the real activity also has another preference (or mix how wish to see the log with how store the log, etc). Or bad experiences in someone with less experience with Git and bad luck with a difficult rebase, etc.

Reply:
> Maybe, apart from the confusing should the review process could need to separate a bit the preferences opinionable.

> That only will end with or do with the reviewer preferences learning each reviewer or debates about what each one thing is more beautiful. Instead of arguments about what could produce errors, maintenability, etc.

> And maiby frustration if both have strong preferences between what the history must be, ideal commits perfect for one preference or near to the real activity also has another preference (or mix how wish to see the log with how store the log, etc). Or bad experiences in someone with less experience with Git and bad luck with a difficult rebase, etc.

This is not something we can solve with documentation. Some reviewers have different preferences than others, and documenting it differently will not affect how they review PRs.

Reply:
I agree with you. Surely it's out of this PR scope, sorry about that.

More than doc for contributors, I mean if there is some review guidelines to avoid that small friction in preferences, that is normal could happen. Maybe it's better avoid a bit the personal preferences to reject/requests/delay PRs.